rust/agents/relayer/src/relayer.rs

```diff
@@ -85,26 +86,33 @@ impl BaseAgent for Relayer {
         let (signed_checkpoint_sender, signed_checkpoint_receiver) =
             watch::channel::<Option<MultisigSignedCheckpoint>>(None);
 
-        let inboxes = self.inboxes();
+        let num_mailboxes = self.core.mailboxes.len();
 
-        let mut tasks = Vec::with_capacity(inboxes.len() + 3);
+        let mut tasks = Vec::with_capacity(num_mailboxes + 1);
```

Comment:
I think this should be `+ 2` because it's:
* num_mailboxes - 1 `run_mailbox` tasks
* checkpoint fetcher
* run_mailbox_sync
* run_interchain_gas_paymaster_sync